lib/kubernetes-deploy/renderer.rb

```diff
@@ -15,6 +15,7 @@ def initialize(msg, parents: [], content: nil, filename:)
       end
     end
     class PartialNotFound < InvalidTemplateError; end
+    attr_reader :template_dir
```

Comment:
This is currently needed inside `split_templates`, since we need to find the renderer associated with a given template directory. This feels like it could be refactored, though. Instead of passing `@template_dirs` around, we could simply pass around the renderer... we would still need this `attr_reader`, but it might read a bit easier.